fuel-crypto/src/secp256/signature.rs

```diff
@@ -137,3 +146,23 @@ impl Signature {
         k1::verify(*self.0, **public_key, message)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::Signature;
+    use test_case::test_case;
+
+    #[test_case(0x00 => 0x00)]
+    #[test_case(0x7E => 0x7E)]
+    #[test_case(0x7F => 0x7F)]
+    #[test_case(0x80 => 0x00)]
+    #[test_case(0xFF => 0x7F)]
+    fn removes_recovery_id(s_byte: u8) -> u8 {
```

Comment:
This test feels like it has wrong layer of abstraction, but I cannot figure out how to test this on a higher level